Make sure we validate simple syntax on service commands

vdemeester opened the pull request:
We ignored errors for simple syntax in `PortOpt` (missed that in the previous migration of this code 🐻). This make sure we don't ignore `nat.Parse` errors.

Test has been migrate too (errors are not exactly the same as before though -_-).

/cc @thaJeztah @AkihiroSuda @icecrime @dnephin @mavenugo @aaronlehmann 

🐸

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

thaJeztah commented on opts/port.go:
```diff
@@ -95,7 +95,10 @@ func (p *PortOpt) Set(value string) error {
 		// short syntax
 		portConfigs := []swarm.PortConfig{}
 		// We can ignore errors because the format was already validated by ValidatePort
```
Need to remove this comment

AkihiroSuda commented on opts/port_test.go:
```diff
@@ -245,6 +245,38 @@ func TestPortOptInvalidComplexSyntax(t *testing.T) {
+	}{
+		{
+			value:         "9999999",
+			expectedError: "Invalid containerPort: 9999999",
```
nit: Probably error strings start with lower characters?

vdemeester replied:
@AkihiroSuda yep, but this errors are from another package so I don't have "here" a way to change that 👼

thaJeztah commented on cli/command/service/opts.go:
```diff
@@ -360,17 +359,6 @@ func (opts *healthCheckOptions) toHealthConfig() (*container.HealthConfig, error
-	portMappings, err := nat.ParsePortSpec(value)
-	for _, portMapping := range portMappings {
-		if portMapping.Binding.HostIP != "" {
-			return "", fmt.Errorf("HostIP is not supported by a service.")
```
As we were discussing; this validation should probably be done somewhere still (don't think `HostIP` is supported for services)

vdemeester replied:
Oh right, I should update `PortOpt` with that.. 👼

vdemeester replied:
And `PortOpt` updated now 👼

thaJeztah commented on opts/port.go:
```diff
@@ -94,11 +94,20 @@ func (p *PortOpt) Set(value string) error {
+		for _, portBindings := range portBindingMap {
+			for _, portBinding := range portBindings {
+				if portBinding.HostIP != "" {
+					return fmt.Errorf("HostIP is not supported by a service.")
```
Perhaps

```
HostIP is not supported for services
```

But that's just a nit (original message was already like this)

dnephin replied:
Since this code isn't in the service package anymore, I think it's wrong to say anything about services. This code could easily be used somewhere else and it would be difficult to notice the error message is wrong.

How about just "HostIP is not supported in %s" or something like that? 

This function is already pretty massive as well. What do you think about making this a separate function?

thaJeztah reviewed (approved):
LGTM

LK4D4 commented:
@vdemeester need rebase, not sure if still actual

vdemeester commented:
@LK4D4 rebased and yeah, still actual 👼

LK4D4 commented:
@vdemeester what do you think about @dnephin suggestion?

vdemeester commented:
@LK4D4 I have to take care of it 👼 😝

vdemeester committed:
Make sure we validate simple syntax on service commands

We ignored errors for simple syntax in `PortOpt` (missed that in the
previous migration of this code). This make sure we don't ignore
`nat.Parse` errors.

Test has been migrate too (errors are not exactly the same as before
though -_-)

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

LK4D4 commented:
ping @dnephin

dnephin reviewed (approved):
LGTM

LK4D4 reviewed (approved):
LGTM
